src/nupic/engine/Network.hpp

```diff
@@ -255,6 +256,14 @@ namespace nupic
     getRegions() const;
 
     /**
+     * Get all links between regions
+     *
+     * @returns A Collection of Link objects in the network
+     */
+    const Collection<Link *>
+    getLinks() const;
```

Comment:
Should this instead be `const Collection<const Link *>` to make sure that links themselves couldn't be altered?